trl/trainer/grpo_trainer.py

```diff
@@ -602,13 +621,23 @@ def _enable_gradient_checkpointing(self, model: PreTrainedModel, args: GRPOConfi
 
         return model
 
+    @profiling_decorator
+    def _get_hidden_states(self, model, input_ids, attention_mask, logits_to_keep=None):
+        # need to unwrap the model to get the hidden states
+        # intended to work with DP, DDP, FSDP
+        unwrapped_model = self.accelerator.unwrap_model(model)
+        hidden_states = unwrapped_model.model(input_ids=input_ids, attention_mask=attention_mask)[0]
```

Comment:
here you get the logits, not the hidden_states, don't you?

You want to use this instead:

```suggestion
        hidden_states = unwrapped_model.model(input_ids=input_ids, attention_mask=attention_mask, output_hidden_states=True).hidden_states
```

Reply:
just need the last hidden state of the model

Reply:
as we call the unwrapped_model.model (it is calling the base model which returns BaseModelOutput)

Reply:
refactored to use last_hidden_state option explicitly